test/controller/confirm.company.controller.spec.unit.ts

```diff
@@ -239,7 +246,36 @@ describe("confirm company tests", () => {
     mockGetObjectionSessionValue.mockImplementation(() => dummyCompanyProfile);
 
     mockGetCompanyEligibility.mockReset();
-    mockGetCompanyEligibility.mockImplementation(() => true);
+    mockGetCompanyEligibility.mockImplementation(() => dummyCompanyEligibility);
+
+    mockLatestGaz1FilingHistoryItem.mockResolvedValueOnce(dummyFilingHistoryItem);
+
+    const response = await request(app).get(OBJECTIONS_CONFIRM_COMPANY)
+      .set("Referer", "/")
+      .set("Cookie", [`${COOKIE_NAME}=123`]);
+
+    expect(mockGetObjectionSessionValue).toHaveBeenCalledTimes(1);
+    expect(response.status).toEqual(200);
+
+    expect(response.text).toContain("00006400");
+    expect(response.text).toContain("14 April 2015");
+  });
+
+  it("should correctly display the latest GAZ1 date from the filing history if company in gaz1, with gaz2 requested", async () => {
+
+    const mockValidAccessToken = retrieveAccessTokenFromSession as jest.Mock;
+
+    beforeEach(() => {
```

Comment:
the before each should be outside of the tests, or if you only need to reset the access token for this test then just remove the beforeEach wrapper